cmd/hc-install/cmd_install.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/hashicorp/go-version"
+	"github.com/mitchellh/cli"
+	"gophers.dev/pkgs/ignore"
+
+	hci "github.com/hashicorp/hc-install"
+	"github.com/hashicorp/hc-install/product"
+	"github.com/hashicorp/hc-install/releases"
+	"github.com/hashicorp/hc-install/src"
+)
+
+type InstallCommand struct {
+	Ui cli.Ui
+}
+
+func (c *InstallCommand) Name() string { return "install" }
+
+func (c *InstallCommand) Synopsis() string {
+	return "Install a HashiCorp product"
+}
+
+func (c *InstallCommand) Help() string {
+	helpText := `
+Usage: hc-install install [options] -version <version> <product>
+
+  This command installs a HashiCorp product.
+  Options:
+    -version  [REQUIRED] Version of product to install.
+    -path     Path to directory where the product will be installed. Defaults
+              to current working directory.
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (c *InstallCommand) Run(args []string) int {
+
+	var (
+		version        string
+		installDirPath string
+	)
+
+	fs := flag.NewFlagSet("install", flag.ExitOnError)
+	fs.Usage = func() { c.Ui.Output(c.Help()) }
+	fs.StringVar(&version, "version", "", "version of product to install")
+	fs.StringVar(&installDirPath, "path", "", "path to directory where production will be installed")
+
+	if err := fs.Parse(args); err != nil {
+		return 1
+	}
+
+	// golang's arg parser is Posix-compliant but doesn't match the
+	// common GNU flag parsing argument, so force an error rather than
+	// silently dropping the options
+	args = fs.Args()
+	if len(args) != 1 {
+		c.Ui.Error(`This command requires one positional argument: <product>
+Option flags must be provided before the positional argument`)
+		return 1
+	}
+	product := fs.Args()[0]
+
+	if version == "" {
+		c.Ui.Error("-version flag is required")
+		return 1
+	}
```

Comment:
As discussed earlier, I'm fine with this behaviour for now. We can still make it optional later and decide later what the right default is and what it means for pre-releases.

@kmoe Do you have any different thoughts here?

Reply:
Yes, I like this approach. Require `-version` for now and we can always make it optional later.